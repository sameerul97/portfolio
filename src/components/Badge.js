import React from 'react'

const ClickableBadge = ({ tag }) => (
  <a href={tag.link} target="_blank" rel="noopener noreferrer">
    <span className="badge rounded-pill bg-secondary me-1 fw-light">{tag.name}</span>
  </a>
)

const Badge = ({ tag, classes }) => (
  <span className={`badge rounded-pill bg-dark me-1 fw-light ${classes}`}>{tag.name}</span>
)

export { ClickableBadge, Badge }
