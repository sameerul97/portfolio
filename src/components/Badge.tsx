import React from 'react'
import { ClickableBadgeprops, BadgeProps } from '../interfaces'

const ClickableBadge = ({ tag }: ClickableBadgeprops) => (
  <a href={tag.link} target="_blank" rel="noopener noreferrer">
    <span className="badge rounded-pill bg-secondary mr-1 p-1 px-2 text-white font-weight-light">{tag.name}</span>
  </a>
)

const Badge = ({ tag, classes }: BadgeProps) => (
  <span className={`badge rounded-pill bg-dark mr-1 p-1 px-2 text-white font-weight-light ${classes}`}>{tag.name}</span>
)

export { ClickableBadge, Badge }
