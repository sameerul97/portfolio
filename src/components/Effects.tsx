// import React, { useEffect, useRef } from 'react'
// import { useThree, useFrame, extend } from '@react-three/fiber'
// // import { EffectComposer } from 'three/examples/jsm/postprocessing/EffectComposer'
// // import { ShaderPass } from 'three/examples/jsm/postprocessing/ShaderPass'
// // import { RenderPass } from 'three/examples/jsm/postprocessing/RenderPass'
// // import { UnrealBloomPass } from 'three/examples/jsm/postprocessing/UnrealBloomPass.js'

// // extend({ EffectComposer, ShaderPass, RenderPass, UnrealBloomPass })

// export function Effects() {
//   const composer = useRef()
//   const bloomPass = useRef()

//   const { scene, gl, size, camera } = useThree()

//   useEffect(() => void composer.current.setSize(size.width, size.height), [size])

//   useFrame(() => {
//     composer.current.render()
//   }, 1)
//   return (
//     <effectComposer ref={composer} args={[gl]}>
//       <renderPass attachArray="passes" scene={scene} camera={camera} />
//       <unrealBloomPass attachArray="passes" ref={bloomPass} args={[0.2, 1.017, 0.021]} />
//     </effectComposer>
//   )
// }

import React from 'react'
import { EffectComposer, DepthOfField, Noise, Vignette } from '@react-three/postprocessing'
import { Bloom } from '@react-three/postprocessing'

export function Effects() {
  return (
    <EffectComposer>
      {/* <DepthOfField focusDistance={0} focalLength={0.02} bokehScale={2} height={480} /> */}
      <Bloom
        intensity={2.0} // The bloom intensity.
        width={1645} // render width
        height={9000} // render height
        kernelSize={5} // blur kernel size
        luminanceThreshold={1.9} // luminance threshold. Raise this value to mask out darker elements in the scene.
        luminanceSmoothing={0.0025} // smoothness of the luminance threshold. Range is [0, 1]
      />
      {/* <Noise opacity={0.02} /> */}
      {/* <Vignette eskil={false} offset={0.1} darkness={1.1} /> */}
    </EffectComposer>
  )
}
