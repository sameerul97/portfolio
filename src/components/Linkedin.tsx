import * as THREE from 'three'
import React, { useRef } from 'react'
import { useGLTF } from '@react-three/drei'
import { GLTF } from 'three-stdlib'
import { Mesh } from 'three'
import { useFrame } from '@react-three/fiber'
import { useSpring, animated } from '@react-spring/three'

type GLTFResult = GLTF & {
  nodes: {
    Curve: THREE.Mesh
  }
  materials: {
    ['SVGMat.001']: THREE.MeshStandardMaterial
  }
}

// export default function Github({ githubHovered }: { githubHovered: boolean }) {
//   const ref = useRef<Mesh>(null!)
//   const { nodes, materials } = useGLTF('/github.glb') as unknown as GLTFResult

//   useFrame((state, delta) => {
//     if (ref.current) {
//       const t = state.clock.getElapsedTime()
//       ref.current.rotation.z += delta / 4
//       ref.current.position.z = THREE.MathUtils.lerp(ref.current.position.z, githubHovered ? 0 : 10, 0.1)
//       ref.current.position.x = THREE.MathUtils.lerp(ref.current.position.x, githubHovered ? -0.003 : 1, 0.1)
//       ref.current.position.y = THREE.MathUtils.lerp(
//         ref.current.position.y,
//         true ? (-0.5 + Math.sin(t)) / 12 : -4.3,
//         0.1
//       )
//     }
//   })

//   return (
//     <group scale={16} dispose={null}>
//       <mesh
//         ref={ref}
//         rotation={[Math.PI / 2, 0, 0]}
//         castShadow
//         receiveShadow
//         geometry={nodes.Curve.geometry}
//         material={materials.SVGMat}
//         scale={12}>
//         <meshPhysicalMaterial color="purple" metalness={0.1} transmission={1} thickness={1} roughness={0.1} />
//       </mesh>
//     </group>
//   )
// }

function getRandomArbitrary(min: number, max: number) {
  return Math.random() * (max - min) + min
}

export default function LinkedIn({ x, y }: { x: number; y: number }) {
  const ref = useRef<Mesh>(null!)
  const { nodes, materials } = useGLTF('/linkedin.glb') as unknown as GLTFResult

  const { position, scale, rotation, opacity } = useSpring({
    to: {
      position: [getRandomArbitrary(-2, 2), getRandomArbitrary(2.5, 4), 2],
      scale: getRandomArbitrary(90, 120),
      rotation: [Math.PI / 2, 0, 3],
      opacity: 0.3,
    },
    from: {
      position: [getRandomArbitrary(-2, 0.5), getRandomArbitrary(-4, -2), 0],
      rotation: [Math.PI / 2, 0, 0],
      scale: getRandomArbitrary(120, 200),
      opacity: 1,
    },
    config: { duration: 4000, mass: 5, tension: 500, friction: 150 },
  })

  return (
    <animated.mesh
      // @ts-ignore
      position={position}
      scale={scale}
      // @ts-ignore
      rotation={rotation}
      ref={ref}
      // rotation={[Math.PI / 2, 0, 0]}
      castShadow
      receiveShadow
      geometry={nodes.Curve.geometry}>
      {/* <boxBufferGeometry attach="geometry" args={[1, 1, 1]} /> */}
      {/* <animated.meshStandardMaterial color="white" opacity={opacity} /> */}

      {/* @ts-ignore */}
      <animated.meshPhysicalMaterial
        color={'#5effff'}
        // @ts-ignore
        shininess={1}
        metalness={1}
        transmission={1}
        thickness={3}
        roughness={0.4}
      />
      {/* <animated.meshPhongMaterial
        color={'white'}
        shininess={1}
        metalness={1}
        roughness={0.1}
        transparent={true}
        // opacity={opacity}
      /> */}
    </animated.mesh>
  )
}

useGLTF.preload('/linkedin.glb')
