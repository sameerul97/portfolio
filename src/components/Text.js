import React from 'react'

const HeaderText = (props) => <h1>{props.text}</h1>

const Text = (props) => {
  if (props.text) return <HeaderText text={props.text} />

  return <p>No text</p>
}

export default Text

export { HeaderText }
