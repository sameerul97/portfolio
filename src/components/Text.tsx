import React from 'react'

const HeaderText = (props: { text: string }) => <h1>{props.text}</h1>

const Text = (props: { text?: string }) => {
  if (props.text) return <HeaderText text={props.text} />

  return <p>No text</p>
}

export default Text

export { HeaderText }
