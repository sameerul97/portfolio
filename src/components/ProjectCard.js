import React from 'react'
import { ClickableBadge, Badge } from './Badge'

const ProjectCard = ({ project }) => {
  return (
    <div className="card border-light p-1">
      <ProjectImage image={project.image} imageCaption={project.name} />

      <ProjectBody>
        <ProjectName name={project.name} />

        {project.projectTags.map((tag) =>
          tag.link ? <ClickableBadge Badge key={tag.id} tag={tag} /> : <Badge Badge key={tag.id} tag={tag} />
        )}
      </ProjectBody>
    </div>
  )
}

const ProjectImage = ({ image, imageCaption }) => <img src={image} className="card-img-top" alt={imageCaption} />

const ProjectName = ({ name }) => <h6 className="card-text text-dark">{name}</h6>

const ProjectBody = ({ children }) => <div className="card-body">{children}</div>

export { ProjectCard, ProjectImage, ProjectName, ProjectBody }
