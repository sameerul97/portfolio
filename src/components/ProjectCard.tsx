import React, { ReactNode } from 'react'
import { ClickableBadge, Badge } from './Badge'
import { ProjectCardProps } from '../interfaces'

const ProjectCard = ({ project }: { project: ProjectCardProps }) => {
  return (
    <div className="card border-light p-1" data-testid={project.id}>
      <ProjectImage image={project.image} imageCaption={project.name} />

      <ProjectBody>
        <ProjectName name={project.name} />

        {project.projectTags.map((tag) =>
          tag.link ? <ClickableBadge key={tag.id} tag={tag} /> : <Badge key={tag.id} tag={tag} />
        )}
      </ProjectBody>
    </div>
  )
}

const ProjectImage = ({ image, imageCaption }: { image: string; imageCaption: string }) => (
  <img src={image} className="card-img-top" alt={imageCaption} />
)

const ProjectName = ({ name }: { name: string }) => <h6 className="card-text text-dark">{name}</h6>

const ProjectBody = ({ children }: { children: ReactNode }) => <div className="card-body">{children}</div>

export { ProjectCard, ProjectImage, ProjectName, ProjectBody }
