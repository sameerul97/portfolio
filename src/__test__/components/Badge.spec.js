import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'
import { screen } from '@testing-library/react'

/* Components */
import Text, { HeaderText } from '../../components/Text'
import { ClickableBadge, Badge } from '../../components/Badge'

Enzyme.configure({ adapter: new Adapter() })

describe('Badge', () => {
  it('should render clickable badge', () => {
    const tag = {
      link: 'https://github.com',
      name: 'Github',
    }

    const wrapper = shallow(<ClickableBadge tag={tag} />)
    const aTag = wrapper.find('a')
    const spanTag = wrapper.find('span')

    expect(aTag.props().href).toBe(tag.link)
    expect(spanTag.text()).toEqual(tag.name)
  })

  it('should render badge', () => {
    const tag = {
      name: 'Docker',
    }

    const wrapper = shallow(<Badge tag={tag} />)
    const spanTag = wrapper.find('span')

    expect(spanTag.text()).toEqual(tag.name)
  })

  it('should render badge with custom CSS class', () => {
    const tag = {
      name: 'Docker',
    }
    const classes = 'bg-white text-dark customClass'
    const wrapper = shallow(<Badge tag={tag} classes={classes} />)

    expect(wrapper.hasClass(classes)).toBe(true)
  })

  it('should render text', () => {
    const text = 'hello world'
    const wrapper = shallow(<Text text={text} />)

    expect(toJson(wrapper)).toMatchSnapshot()
    expect(wrapper.find(HeaderText))
    expect(wrapper.find(HeaderText).prop('text')).toBe(text)
  })

  it('should render no text', () => {
    const text = 'No text'
    const wrapper = shallow(<Text />)
    const paragraph = wrapper.find('p')

    expect(toJson(wrapper)).toMatchSnapshot()
    expect(paragraph.text()).toEqual(text)
  })
})
