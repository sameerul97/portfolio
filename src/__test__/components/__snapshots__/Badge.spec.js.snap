// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge should render no text 1`] = `
<p>
  No text
</p>
`;

exports[`Badge should render text 1`] = `
<HeaderText
  text="hello world"
/>
`;

exports[`Paragraph should render no text 1`] = `
<p>
  No text
</p>
`;

exports[`Paragraph should render text 1`] = `
<HeaderText
  text="hello world"
/>
`;
